### PR TITLE
messenger dao`s implemented v 1.0

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 1]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Comment create(Comment entity) {
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:17]:
```suggestion
    public Comment create(Comment comment) {
```

File: src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java [hunk 2]
```diff
@@ -12,21 +15,64 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
...
     @Override
     public Message create(Message entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            session.save(entity);
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java:24]: ?

File: src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java [hunk 3]
```diff
@@ -12,21 +15,64 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Message create(Message entity) {
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java:17]:
```suggestion
    public Message create(Message message) {
```

File: src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java [hunk 4]
```diff
@@ -12,21 +15,64 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
...
+        } catch (Exception e) {
+            throw new RuntimeException("Can`t find by id: " + id, e);
+        }
     }
 
     @Override
     public List<Message> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Message> getAllMessageQuery = session.createQuery("from Message", Message.class);
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java:51]:
```suggestion
            Query<Message> getAllMessageQuery = session.createQuery("FROM Message", Message.class);
```

File: src/main/java/core/basesyntax/dao/impl/MessageDetailsDaoImpl.java [hunk 5]
```diff
@@ -11,11 +13,33 @@ public MessageDetailsDaoImpl(SessionFactory sessionFactory) {
...
     @Override
     public MessageDetails create(MessageDetails entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            session.save(entity);
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/MessageDetailsDaoImpl.java:22]: ?

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 6]
```diff
@@ -12,16 +15,43 @@ public SmileDaoImpl(SessionFactory sessionFactory) {
...
     @Override
     public Smile create(Smile entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            session.save(entity);
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:24]: ?

File: src/main/java/core/basesyntax/model/Comment.java [hunk 7]
```diff
@@ -1,15 +1,24 @@
...
+import javax.persistence.Table;
 
+@Entity
+@Table(name = "comments")
 public class Comment {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String content;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/model/Comment.java:20]: OneToMany or ManyToMany? Also, it's a bad practice to set Eager fetch type for toMany relations.

**Review by vovasalyha — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 8]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            session.save(entity);
```
[the remark this answers is not in the document]
Comment (Skarbey) [reply src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java]: becouse i test it before and forgot

**Comment by Skarbey** [comment]
Thank you!

File: src/main/java/core/basesyntax/model/Comment.java (hunk 7 above)
Comment (Skarbey) [reply src/main/java/core/basesyntax/model/Comment.java]: Users have many comments
Comments have many users?

**Pull request closed without merging**